cartographer/pose_graph/proto/cost_function.proto

```diff
@@ -0,0 +1,68 @@
+// Copyright 2018 The Cartographer Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+import "cartographer/pose_graph/proto/node.proto";
+import "cartographer/transform/proto/transform.proto";
+
+package cartographer.pose_graph.proto;
+
+message RelativePose2D {
+  NodeId first = 1;
+  NodeId second = 2;
+
+  transform.proto.Rigid2d first_from_second = 3;
```

Comment:
Is this Cartographers notation?
In Tango we had second_T_first to denote a transformation from first to second; this has the advantage that we can chain these pretty easily:

third_T_first = third_T_second * second_T_first

This notation is also pretty common in robotics.

Reply:
Let's change to `first_T_second`, sure. As long as it can be chained.